client/src/templates/Challenges/utils/build.js

```diff
@@ -44,7 +44,15 @@ const composeFunctions = (...fns) =>
 
 function buildSourceMap(challengeFiles) {
   // TODO: rename sources.index to sources.contents.
-  const source = challengeFiles.reduce(
+  // Make sure index.html is always first
+  let fixedChallengeFiles = challengeFiles;
+  if (
+    challengeFiles.length === 2 &&
+    challengeFiles[1].fileKey === 'indexhtml'
+  ) {
+    fixedChallengeFiles = [...challengeFiles].reverse();
+  }
+  const source = fixedChallengeFiles.reduce(
```

Comment:
```suggestion
  // Make sure index.html is always first
  const source = sortChallengeFiles(challengeFiles).reduce(
```
I think we'll always want the `index.html` to be first, even when we're looking at steps that have `js` or `jsx` files.  `sortChallengeFiles` can be found in utils/sort-challengefiles.js

Reply:
Sorry, that won't work.  I'd forgotten that the `ext` property gets transformed, so `sortChallengeFiles` doesn't know what to do with these.  I'll make that function more robust and then this should be okay.